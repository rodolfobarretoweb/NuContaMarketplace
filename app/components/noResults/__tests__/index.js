import React from 'react';
import renderer from 'react-native-test-utils';
import checkPropTypes, { assertPropTypes } from 'check-prop-types';
import NoResults from '../index';

describe('NoResults component', () => {
  it('renders snapshot correctly', () => {
    const noResults = renderer(<NoResults text="No results" />);
    expect(noResults.toJSON()).toMatchSnapshot();
  });

  it('fails with wrong prop type on `text` prop', () => {
    expect(
      checkPropTypes(NoResults.propTypes, { text: false }, 'prop', 'NoResults')
    ).toContain('Invalid prop `text` of type `boolean` supplied to `NoResults`, expected `string`.');
  });
});
