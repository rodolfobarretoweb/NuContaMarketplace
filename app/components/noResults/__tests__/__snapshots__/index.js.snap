// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResults component renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    No results
  </Text>
</View>
`;
